qiskit/dagcircuit/dagcircuit.py

```diff
@@ -996,8 +996,11 @@ def threeQ_or_more_gates(self):
                 three_q_gates.append(node)
         return three_q_gates
 
-    def longest_path(self):
+    def deepest_path(self):
         """Returns the longest path in the dag as a list of DAGNodes."""
+        # Note: naming for a dagcircuit is the following
+        #       deepest path: path with largest number of operations
+        #       longest path: path which takes longest to execute
```

Comment:
This DAG does not have the notion of time, so I'm not fully sure if I'm okey with this renaming/comment. If renaming, because it's user-facing API, the previous name should be deprecated.

Reply:
Yes, the DAG should not have a notion of time, but the naming of the dag properties and the transpiler analysis passes has to be consistent, ideally right from the point where `nx.dag_longest_path()` is being called. I have added a depracation warning to `longest_path()`.